devicetypes/smartthings/zwave-mouse-trap.src/zwave-mouse-trap.groovy

```diff
@@ -0,0 +1,258 @@
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name: "pestControl", type: "generic", width: 6, height: 4) {
+			tileAttribute("device.pestControl", key: "PRIMARY_CONTROL") {
+				attributeState("idle", label: 'IDLE', icon: "st.contact.contact.open", backgroundColor: "#00FF00")
+				attributeState("trapRearmRequired", label: 'TRAP RE-ARM REQUIRED', icon: "st.contact.contact.open", backgroundColor: "#00A0DC")
+				attributeState("trapArmed", label: 'TRAP ARMED', icon: "st.contact.contact.open", backgroundColor: "#FF6600")
+				attributeState("pestDetected", label: 'PEST DETECTED', icon: "st.contact.contact.open", backgroundColor: "#FF6600")
+				attributeState("pestExterminated", label: 'PEST EXTERMINATED', icon: "st.contact.contact.closed", backgroundColor: "#FF0000")
+			}
+		}
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: '', action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		standardTile("configure", "device.configure", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "configure", label: '', action: "configuration.configure", icon: "st.secondary.configure"
+		}
+		main "pestControl"
+		details(["pestControl", "battery", "refresh", "configure"])
+	}
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	log.debug "ping() called"
+	refresh()
+}
+
+def refresh() {
+	log.debug "sending battery refresh command"
+	def cmds = []
+	cmds << zwave.batteryV1.batteryGet().format()
+	cmds << zwave.manufacturerSpecificV2.manufacturerSpecificGet().format()
+	cmds << zwave.notificationV3.notificationGet(notificationType: 0x13).format()
+	cmds << zwave.sensorBinaryV2.sensorBinaryGet(sensorType: zwave.sensorBinaryV2.SENSOR_TYPE_CO).format()
+	return delayBetween(cmds, 2000)
+}
+
+def parse(String description) {
+	def result = []
+	log.debug "desc: $description"
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug "parsed '$description' to $result"
+	return result
+}
+
+def installed() {
+	log.debug "installed()"
+	// Device-Watch simply pings if no device events received for 8h 6min(checkInterval)
+	sendEvent(name: "checkInterval", value: 24 * 60 * 60 + 6 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	initialize()
+}
+
+def updated() {
+	log.debug "updated()"
+	// Device-Watch simply pings if no device events received for 8h 6min(checkInterval)
+	sendEvent(name: "checkInterval", value: 24 * 60 * 60 + 6 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	runIn(12, "initialize", [overwrite: true, forceForLocallyExecuting: true])
+}
+
+def initialize() {
```

Comment:
This doesn't give you any guarantee that you _ever_ send the configuration commands. 

If you don't fail to send them on startup (due to the device going to sleep unprompted I'm guessing), they will never be sent. If the device is sleepy and it doesn't receive them initially, it definitely won't receive them after 12s. I'd recommend just removing this whole method in favor of just calling it once on install, then checking in the wakeup handler and possibly sending configuration commands there.